### PR TITLE
feat(WEBRTC-215): add ondemand credential resource

### PR DESCRIPTION
**Pull request by DeividVeloso**
[WEBRTC-215](https://telnyx.atlassian.net/browse/WEBRTC-215)

Added new endpoints inside SDK Node to expose the telephony_credentials paths.
 - https://api.telnyx.com/v2/telephony_credentials
 - https://api.telnyx.com/v2/telephony_credentials/{id}/token

To run the tests correctly you will need to  add the prop `street_address: '311 W Superior Street'` on test/resources/Addresses.spec.js at line 21.

Notes: 

Before merge on master wait for PR #48 be approve.

### CODE REVIEW

File: test/resources/TelephonyCredentials.spec.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      expect(response.data).to.have.property('id');
+      expect(response.data).to.have.property('name');
+      expect(response.data).to.have.property('record_type');
+      expect(response.data).to.have.property('resource_id');
+      expect(response.data).to.have.property('sip_password');
+      expect(response.data).to.have.property('sip_username');
+    }
+
+    it('Sends the correct request', function () {
+      return telnyx.telephonyCredentials.create(credentialCreateData).then(responseFn);
```
Comment (SuaYoo) [inline test/resources/TelephonyCredentials.spec.js:30]: I don't see this being mocked in the PR, is the test running against the actual API?

File: test/resources/Addresses.spec.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ var addressData = {
   last_name: 'Foster',
   locality: 'Chicago',
   postal_code: '2904',
+  street_address: '311 W Superior Street'
```
Comment (SuaYoo) [inline test/resources/Addresses.spec.js:21]: Looks like you'll want to either branch from https://github.com/team-telnyx/telnyx-node/pull/49 or omit the commit from this PR, or else you'll run into merge conflicts

File: test/resources/TelephonyCredentials.spec.js (hunk 1 above)
Comment (DeividVeloso) [reply test/resources/TelephonyCredentials.spec.js:30]: This project has an API mock that you need to run separately and concurrently:

You can check here https://github.com/team-telnyx/telnyx-node#development

File: test/resources/Addresses.spec.js (hunk 2 above)
Comment (DeividVeloso) [reply test/resources/Addresses.spec.js:21]: I will remove from this PR and wait for the #48

Comment (DeividVeloso) [reply test/resources/Addresses.spec.js]: Done

**Review by SuaYoo — APPROVED** [review]

**Pull request merged**